src/frontend/src/scheduler/worker_node_manager.rs

```diff
@@ -220,10 +220,16 @@ impl WorkerNodeManager {
 
     pub fn remove_streaming_fragment_mapping(&self, fragment_id: &FragmentId) {
         let mut guard = self.inner.write().unwrap();
-        guard
+        if guard
             .streaming_fragment_vnode_mapping
             .remove(fragment_id)
-            .unwrap();
+            .is_none()
+        {
+            tracing::warn!(
+                "Streaming vnode mapping not found for fragment {}",
+                fragment_id
+            );
+        }
```

Comment:
Do we know a (reasonable) situation that can possibly trigger this? May comment here if so, otherwise, I tend to let it panic.

Reply:
Not sure, let @chenzl25 comment on it more, don't quite understand the FIXME in this code.

Reply:
We can also return a scheduler error which propagates to the frontend. I think recovery test is able to handle that. Let me change it.

Reply:
We used panic here to verify that the mechanism of notification works well. 🤔 Seems like it panicked when trying to remove the old fragment mapping for `sink into table`. I can't come with a scene will trigger it, let me take another look. It's best to keep the panic logic.

Reply:
Hmm okay, I will let the maincron tests fail for now then. We can fix this in a separate PR. Will revert the changes later

Reply:
Okay, I'll try to reproduce it locally and see what's happening.

Reply:
I have identify the root cause, will try to rise a PR to fix it.

Reply:
Fixed in #15909 .